fdb-record-layer-jmh/src/jmh/java/com/apple/foundationdb/record/benchmark/lucene/MailDataLoader.java

```diff
@@ -0,0 +1,251 @@
+import com.apple.foundationdb.record.benchmark.BenchmarkRecordStore;
+import com.apple.foundationdb.record.provider.common.DynamicMessageRecordSerializer;
+import com.apple.foundationdb.record.provider.foundationdb.FDBDatabase;
+import com.apple.foundationdb.record.provider.foundationdb.FDBDatabaseFactory;
+import com.apple.foundationdb.record.provider.foundationdb.FDBRecordContext;
+import com.apple.foundationdb.record.provider.foundationdb.FDBRecordStore;
+import com.apple.foundationdb.record.provider.foundationdb.FDBRecordStoreBase;
+import com.apple.foundationdb.record.provider.foundationdb.keyspace.KeySpacePath;
+import org.apache.commons.math3.util.Pair;
+
+import javax.annotation.Nonnull;
+import javax.mail.BodyPart;
+import javax.mail.Header;
+import javax.mail.Session;
+import javax.mail.internet.MimeMessage;
+import javax.mail.internet.MimeMultipart;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Enumeration;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * A Data loader that reads test data from a directory and writes it to the javax.mail structure.
+ */
+public class MailDataLoader {
+
+    private final RecordMetaData metaData;
+    private final MboxDistribution distribution;
+    private final int batchSize;
+
+    private final String mailDir;
+
+    public MailDataLoader(@Nonnull RecordMetaData metaData,final MboxDistribution distribution, final int batchSize, final String mailDir) {
+        this.metaData = metaData;
+        this.distribution = distribution;
+        this.batchSize = batchSize;
+        this.mailDir = mailDir;
+    }
+
+    public void loadData(@Nonnull final FDBDatabase fdbDb,
+                         @Nonnull final KeySpacePath ksPath) throws Exception {
+        Session session = Session.getDefaultInstance(System.getProperties(), null);
+
+        try (final Stream<Path> fileStream = Files.list(Path.of(mailDir))) {
+            final List<Pair<TestMailProto.MessageState, TestMailProto.TextSearch>> loadedData = fileStream.map(path -> generateNext(session, path.getFileName().toString())).collect(Collectors.toList());
+
+            //TODO(bfines) multithreading this might make sense
+            List<Pair<TestMailProto.MessageState, TestMailProto.TextSearch>> batch = new ArrayList<>(batchSize);
+            for (var elem : loadedData) {
+                batch.add(elem);
+                if ((batch.size() % batchSize) == 0) {
+                    insertBatch(fdbDb, ksPath, batch);
+                    batch.clear();
+                }
+            }
+            if (!batch.isEmpty()) {
+                insertBatch(fdbDb, ksPath, batch);
+            }
+        }
+    }
```

Comment:
This is w/ lucene index? Can I call Lucene optimize on this index? Say after load and before a read test or after a big batch of loadings? Is that possible?